ui/apps/platform/src/Containers/Policies/Wizard/Step2/PolicyBehaviorForm.tsx

```diff
@@ -36,35 +39,46 @@ type PolicyBehaviorFormProps = {
 
 function PolicyBehaviorForm({ hasActiveViolations }: PolicyBehaviorFormProps) {
     const { values, setFieldValue, setValues } = useFormikContext<ClientPolicy>();
+    const [lifeCycleChange, setLifeCycleChange] = useState<{
+        lifecycleStage: LifecycleStage;
+        isChecked: boolean;
+    } | null>(null);
+
     const hasEnforcementActions =
         values.enforcementActions?.length > 0 &&
         !values.enforcementActions?.includes('UNSET_ENFORCEMENT');
     const [showEnforcement, setShowEnforcement] = React.useState(hasEnforcementActions);
 
     function onChangeLifecycleStage(lifecycleStage: LifecycleStage, isChecked: boolean) {
-        /*
-         * Set all changed values at once, because separate setFieldValue calls
-         * for lifecycleStages and eventSource cause inconsistent incorrect validation.
-         */
-        const changedValues = { ...values };
-        if (isChecked) {
-            changedValues.lifecycleStages = [...values.lifecycleStages, lifecycleStage];
+        if (values?.id) {
```

Comment:
For my understanding, not necessarily requesting a change, does optional chaining imply there is a chance that `values` are null or undefined?

Here is an example of how PolicyPage prevents a nullish policy object:
```js
                .catch((error) => {
                    setPolicy(initialPolicy);
```